data/store/mongo/mongo.go

```diff
@@ -94,11 +90,9 @@ func New(logger log.Logger, environmentReporter environment.Reporter, config *Co
 		return nil, app.ExtError(err, "mongo", "unable to determine build info")
 	}
 
-	if !environmentReporter.IsTest() {
-		if !buildInfo.VersionAtLeast(3) {
-			session.Close()
-			return nil, app.Errorf("mongo", "unsupported mongo build version %s", strconv.Quote(buildInfo.Version))
-		}
+	if !buildInfo.VersionAtLeast(3) {
```

Comment:
I couldn't find a way to explicitly test this. However, I can confirm this code does work as expected since it failed on Travis when it was using a version 2.* of Mongo (until I updated the Travis config to use a newly install version 3.* of Mongo).